amxmodx/amx.cpp

```diff
@@ -4173,3 +4172,45 @@ int AMXAPI amx_GetStringOld(char *dest,const cell *source,int use_wchar)
   dest[len]='\0';     /* store terminator */
   return AMX_ERR_NONE;
 }
+
+
+#include <chrono>
+#include <amxmodx.h>
+#include <CPlugin.h>
+
+int AMXAPI amx_ExecPerf(AMX* amx, cell* retval, int index)
+{
+    if (amx && retval)
+    {
+        CPluginMngr::CPlugin* perf_Plug = g_plugins.findPluginFast(amx);
+        if (perf_Plug && perf_Plug->isDebug())
+        {
+            char perf_funcname[512];
+            amx_GetPublic(perf_Plug->getAMX(), index, perf_funcname);
+            if (strlen(perf_funcname) == 0)
+                sprintf(perf_funcname, "Unknown_ID%d", index);
+
+            const char* perf_plugname = perf_Plug->getName();
+            if (!perf_plugname || perf_plugname[0] == '\0')
+                perf_plugname = "Unknown_plugin";
+
+            using std::chrono::high_resolution_clock;
+            using std::chrono::duration_cast;
+            using std::chrono::duration;
+            using std::chrono::milliseconds;
+
+            auto t1 = high_resolution_clock::now();
+            int err = amx_Exec(amx, retval, index);
+            auto t2 = high_resolution_clock::now();
+
+            auto ms_int = duration_cast<milliseconds>(t2 - t1);
+            auto ms_integer = (int)ms_int.count();
+            if (ms_integer > 1)
```

Comment:
Please make this configurable, via cvar or command-line or something. 1ms is probably okay since that's quite a big chunk of the frame budget, but it'd be best to have the server operator have some control over it (and maybe express the units in microseconds instead?)

Reply:
At current time bigger than 1 ms (2+ms) it very big time for one function, yes I think need make it configurable

Reply:
@dvander why better, add new cvar to amxx.cfg file?

Reply:
Pretty much any method would work, I think. The exact method doesn't matter as long as it's easy for server operators to change.

Reply:
I can do it from PC, at current time I used phone for browsing GitHub

Reply:
@dvander you can review new update ? 

https://github.com/alliedmodders/amxmodx/pull/1019/commits/5cb6fa0374289ae6bbac75348a456872b127151d

Now added cvar.